bigquery_etl/query_scheduling/task.py

```diff
@@ -418,6 +428,16 @@ def _duplicate_dependency(task):
                             dependencies.append(task)
                         break  # stop after the first match
 
+        # adjust submission_date parameter based on whether upstream tasks have date_partition offsets
+        date_partition_offsets = [
+            dependency.date_partition_offset
+            for dependency in dependencies
+            if dependency.date_partition_offset
+        ]
+
+        if len(date_partition_offsets) > 0:
+            self.date_partition_offset = min(date_partition_offsets)
```

Comment:
I think it's very likely that we'll add new tasks that depend on a table with offset and forget to specify an explicit offset. So I think it's desirable to infer offset and use the most conservative value by default as we're doing here.

I think your suggestion of being able to override with an explicit offset is a good idea.

I see this as very similar to how we determine dependencies in this repo. If you specify dependencies in the metadata, then the tooling uses that value; otherwise, we infer dependencies through tooling.

Reply:
cc @relud re implications for VPN-related tasks.

Reply:
> use the most conservative value by default as we're doing here

If we want to use the most conservative value then I think we'd need to use `max()` instead of `min()` since the values are negative.

Reply:
> I don't think this will work for the VPN-related tasks using offsets, because for those we actually want dependent tasks to run without an offset unless they explicitly have an offset themselves

It turns out there are currently no such VPN-related downstream dependent tasks that don't use the same offset.